pkg/build/api/validation/validation.go

```diff
@@ -31,7 +31,7 @@ func ValidateBuildUpdate(build *buildapi.Build, older *buildapi.Build) fielderro
 	allErrs = append(allErrs, ValidateBuild(build)...)
 
 	if !kapi.Semantic.DeepEqual(build.Spec, older.Spec) {
-		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec", build.Spec, "spec is immutable"))
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec", "build "+build.ObjectMeta.Name, "spec is immutable"))
```

Comment:
for normal fields, there's minimal context as well... I would hope whatever rolls up the errors tells you the overarching object that failed... if not, that's probably the right place to put identifying info, not in the guts of a field error